### ISSUE TITLE
Filter lines matching regex group/s into new file - new command?

### ISSUE THREAD

**Issue by doubleell** [qcz/vscode-text-power-tools#36]
At the moment the command "Filter lines matching a regex (grep) into new file" copies the whole line where the match occurred.

Is it possible to just copy the capture group/s. I'm using this regex command: (A1_.*?)(?:[,\s])
For the example lines below;
	btfss	A1_VoltageFlags,LowVolts
	tstfsz	A1_ModeTimeout				; check timeout counter

The whole line is copied, I just want the first capture group $1 ie. A1_VoltageFlags or  A1_ModeTimeout 

I can kind of get around this at the moment by using the regex in the vscode search box, then select all Alt-Enter and then running the "Filter lines matching a regex (grep) into new file" command. Although this still gives me whitespace and the trailing comma. If there were another input box to select the output based on the group number ie. $1, $2 etc this would be great!

**Comment by qcz** [comment]
The `Extract information from text` and `Extract information from text into a new editor` does exactly this. First you specify the regex command, and then you need to specify the output, which can contain capture groups.

See it in action here: https://github.com/qcz/vscode-text-power-tools#extracting-information-in-action

**Comment by doubleell** [comment]
Hi, thanks that's exactly what I wanted! I should've visited the help first, anyway thank you :-)

**Comment by doubleell** [comment]
Hi,

Thanks, I used the "extract information from text into new editor", is it
possible to insert new lines in the output? I'm using a capture group from
regex ie.$1 and I format as; void $1(void) {};
but this is a single line, I tried void $1(void) \n {\n };\n
That just puts the literal characters \n in the output.

I wanted the output below, where $1 is an extracted function name;
void $1(void)
{
};

Regards,
Allan.
